lib/company.js

```diff
@@ -0,0 +1,86 @@
+'use strict'
+
+const Promise = require('bluebird')
+
+/**
+ * Company - Used for communication with intercomClient's company object
+ * @class
+ * @author Ryan Kahn
+ */
+class Company {
+
+  /**
+   * Constructor for Company
+   * @param {Orion} orion - Instance of Orion which is used for the intercom client and canUseIntercom methods
+   */
+  constructor (orion) {
+    this.orion = orion
+    if (!this.orion.canUseIntercom()) { return }
+  }
+
+  /**
+   * Helper method to wrap all the intercomClient.companies calls around canUseIntercom
+   * @param {String} method
+   * @returns {Promise} - Returns success if intercom is disabled,
+   * otherwise returns results of call to intercom client
+   * @private
+   */
+  _wrap (method) {
+    if (!this.orion.canUseIntercom()) {
+      return Promise.resolve()
+    }
+    var args = Array.prototype.slice.call(arguments)
+    args.shift()
+    return this.orion.intercomClient.companies[method].apply(this.orion.intercomClient.companies, args)
+  }
+
+  /**
+   * Upsert a company object, passes data through to intercom
+   * Possible parameters: https://developers.intercom.io/reference#create-or-update-company
+   * @param {Object} companyParams
+   * @param {Object} companyParams.company_id - Company unique identifier
+   * @param {Object} companyParams.name - Company name
+   * @returns {Promise} - Returns success if intercom is disabled,
+   * otherwise returns results of create command in intercom
+   */
+  create (companyParams) {
+    return this._wrap('create', companyParams)
+  }
+
+  /**
+   * List companies by a tag or segment
+   * Possible parameters: https://developers.intercom.io/reference#list-by-tag-or-segment
+   * @param {Object} query
+   * @returns {Promise<Array>} - Returns success if intercom is disabled,
+   * otherwise returns list of companies
+   */
+  listBy (query) {
+    return this._wrap('listBy', query)
+      .get('body')
+  }
+
+  /**
+   * List all companies
+   * @returns {Promise<Array>} - Returns success if intercom is disabled,
+   * otherwise returns list of companies
+   */
+  list () {
```

Comment:
this isn't a typical "model" if you'd like I could rename this to `companies` instead of `company`